src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -161,7 +161,7 @@ public async Task<BlobIdentifierWithBlocks> UploadLogToBlobStore(Stream blob, st
         public async Task<(DedupIdentifier dedupId, ulong length)> UploadAttachmentToBlobStore(bool verbose, string itemPath, Guid planId, Guid jobId, CancellationToken cancellationToken)
         {
             var (dedupClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance
-                    .CreateDedupClientAsync(verbose, (str) => Trace.Info(str), this._connection, cancellationToken);
+                    .CreateDedupClientAsync(verbose, (str) => Trace.Info(str), this._connection, 0 /* use default */, cancellationToken);
```

Comment:
if you have `0` mean call `GetDedupStoreClientMaxParallelism()` then you could put zeroes everywhere and this caller would also use the variable

Reply:
I need to figure out how options/overrides are passed to the agent.  The current override mechanism is only available for pipeline tasks.

Reply:
You need the agent context to call that, otherwise I'd assume we'd just internalize the call since this is the only place we don't call GetDedupStoreClientMaxParallelism() first.

Reply:
why not make `DefaultDedupStoreClientMaxParallelism` public and use it directly here? Not a big fan of using 0 to interpret to default

Reply:
I've hooked this up to agent configuration now.  @kutsen99 the reason 0 is a good value is because it is default(int) and when it is read from config and is not specified it will default to 0.  It also keeps the actual default value opaque to callers.

Reply:
The problem is the callers will have to go through the implementation to be able to know that 0 means the default. Better add a comment to the parameter definition.

Reply:
That's not the problem, that's the feature.  The whole purpose of this method is to encapsulate setting the parallelism in a single piece of code and not expose the actual value to clients.  Sure a piece of code could hard code it explicitly but every where it is used the actual piece of code should not know the value and it is passing it through from some config setting or environment variable.  I will add a comment though.